app/models/solr_document.rb

```diff
@@ -28,6 +28,18 @@ class SolrDocument
 
   use_extension( Hydra::ContentNegotiation )
 
+  def visibility
+    if rights_override?
+      @visibility ||= VisibilityService.visibility_value(self[Solrizer.solr_name('rights_override')].first)
+    else
+      super
+    end
+  end
+
+  def rights_override?
+    !self[Solrizer.solr_name('rights_override')].nil? && !self[Solrizer.solr_name('rights_override')].first.blank?
```

Comment:
@mcritchlow - The newer syntax is really nice.  Overall this PR looks 👍 .  Would you like me to merge it or wait for Longshou to update it using fetch option?